### ISSUE TITLE
Passwords too complex

### ISSUE THREAD

**Issue by Mordalo** [nilsteampassnet/TeamPass#1257]
This is a fresh install of 2.1.25 on a Windows 2008R2 server using PHP 7.0, MySQL 5.5 and IIS7 (not Apache).  When trying to create an object, as I enter the password the complexity indicator at the bottom increases from Very Weak to Very Heavy.  Once I enter a password that exceeds the "Very Heavy" criteria and try to save the object, it throws an error "SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data.  If I begin dissecting the password and remove one character at a time until that indicator drops back under "Very Heavy" then it'll let me save without error. 

Not that I think creating rediculously complex passwords is useful, but I have existing passwords that I didn't create that I'd like to store.  

As an example: If I try to save the password .@dm1n1$t@fFP@$$ I get the JSON error.  But if I change it to .@dm1n1$t@fFP it saves just fine.  

![omfg](https://cloud.githubusercontent.com/assets/18295241/14329266/e845fc18-fbf7-11e5-808c-51c32b8a1e66.png)

**Comment by Mordalo** [comment]
The error also occurs if I save the object with a less complex password, then attempt to update it with the more complex characters.  Also, to verify that it wasn't a length restriction, the password .@dm1n1$t@fFP@$$ won't save but .adm1n1stafFPass (note the period at the beginning) will.

**Comment by nilsteampassnet** [comment]
Just tried with editing an existing one and creating a new one and couldn't reproduce (on last development branch commit)
![2016-04-06 21_41_48-teampass](https://cloud.githubusercontent.com/assets/1197546/14330349/7d7321d8-fc40-11e5-9481-a8340ffc3f2e.png)
What could be interesting is that you enable DeveloperTools of your browser and that you copy here the error sent back from the server from the console.

**Comment by Mordalo** [comment]
Since the dev tools have quite a bit of options, is there a particular set of information you are curious about?  Here is the console output:

Error: SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data functions.js:256:5
This site makes use of a SHA-1 Certificate; it's recommended you use certificates with signature algorithms that use hash functions stronger than SHA-1.[Learn More] teampass.redacted.com
TypeError: data is undefined
AjouterItem/<()
 index.php:2162
jQuery.Callbacks/fire()
 jquery.js:3048

jQuery.Callbacks/self.fireWith()
 jquery.js:3160
done()
 jquery.js:8235
.send/callback()
 jquery.js:8778


![debug1](https://cloud.githubusercontent.com/assets/18295241/14333887/22761128-fc0e-11e5-947f-c17785ab5665.png)

**Comment by Mordalo** [comment]
Here is the Network output (I cleared the console just before I hit the Save button so I could capture JUST the submission and response):

Error:
Query: INSERT INTO `teampass_items` (`label`,`description`,`pw`,`pw_iv`,`email`,`url`,`id_tree`,`login`,`inactif`,`restricted_to`,`perso`,`anyone_can_modify`,`complexity_level`) VALUES ('Test4', '', '3eb6d4b79cacd5985b1c90c35e253f49', 'e7a1c14172a62e54fa4d3318074edc47', '', '', '14', '', '0', '', '0', '0', '100')



![debug2](https://cloud.githubusercontent.com/assets/18295241/14334045/0ad85dae-fc0f-11e5-90c3-183b558e2313.png)

**Comment by Mordalo** [comment]
I have an older edition running on a Ubuntu instance that does not have this issue.  So this is either a version specific issue (which I seriously doubt since I don't see anyone else observing this) or (the more likely) something specific to my environment. I'd love to identify the issue so that others could benefit but I don't have the skill to adequately identify the root cause.

**Comment by claumagagnotti** [comment]
Hi, I have the same problem and I running it on CentOS7x64. Issue #1264

**Comment by nilsteampassnet** [comment]
take a look if fields pw and pw_iv have the type "text" inside table "items"

**Comment by Mordalo** [comment]
They do:

![text_fields](https://cloud.githubusercontent.com/assets/18295241/14442183/93378226-fff5-11e5-8d09-3266b821ae9d.png)

**Comment by claumagagnotti** [comment]
Hi Mordalo, Have you solved the problem?

**Comment by Mordalo** [comment]
I have not. I'm not sure where to go from here honestly.

**Comment by claumagagnotti** [comment]
Me too...I spent a lot of time trying to solve it but I couldn't...Im gonna drop it. Regards.

**Comment by Mordalo** [comment]
What I have been doing is creating 2 entries. "username - Part 1" and "username - Part 2" and splitting the passwords in half.  Since it's only for passwords that were not generated by Teampass, I haven't had to do too many.  It's mainly my encryption keys that need this treatment since they tend to be quite long and complex. Still not optimal, but it's a workaround.

**Comment by scbnmoody** [comment]
Also getting this error when using passwords longer than 15 chatacters. Installed was a fresh one with the current version.

**Comment by Mordalo** [comment]
claumagagnotti and scbnmoody, are your teampass instances running in a Windows WAMP stack, or Linux LAMP?  Just curious to see if there's a common theme.

**Comment by scbnmoody** [comment]
WAMP setup for mine. Its pretty much all defaults, nothing too customized

**Comment by Mordalo** [comment]
interesting... whose WAMP stack did you use? Bitnami, AMPPS, MS Web Platform, XAMPP, WampServer or some other derivative? Mine is running on MS Web Platform 3.0.

**Comment by scbnmoody** [comment]
I am using this package http://www.wampserver.com/en/
Right now im using PHP 5.6.16, Apache 2.4.17 and MySQL 5.7.9

**Comment by scbnmoody** [comment]
Just for the heck of it I installed the WAMP 32bit version and a fresh teampass 2.1.25.2 to see if it made a difference and got the same results.

**Comment by scbnmoody** [comment]
Huh, just installed Bitnami XAMPP and now its not giving me the error on a test install. Now lets flip over my production one and see how it reacts.

**Comment by Mordalo** [comment]
So, there may be something in the default config that's different in the bitnami stack.  The question is, which config(s) is/are different?

**Comment by scbnmoody** [comment]
Rolled over my production Teampass over to Bitnami and that issue seems to be resolved as well as having much better performance. I sadly dont have time to go through the two stacks and see what the difference is but at least I have a solution now.

**Comment by TuxThePenguin76** [comment]
This is still an issue with 2.1.26 on Ubuntu 16.04 LAMP.

**Comment by WolfgangEibner** [comment]
I got the same error with a complex password. By trying the shown SQL in mysql cli I got: "ERROR 1406 (22001): Data too long for column 'complexity_level' at row 1".

The column complexity_level of table teampass_items is only varchar(2) but the complexity at such passwords lead to 100 - which can't be inserted.

So... 
`ALTER table teampass_items MODIFY complexity_level VARCHAR(3);`
should solve that problem.
